sched/clock/clock_gettime.c

```diff
@@ -58,15 +58,15 @@ static clock_t clock_process_runtime(FAR struct tcb_s *tcb)
 
   group = tcb->group;
 
-  flags = spin_lock_irqsave(NULL);
+  flags = spin_lock_irqsave(&group->lock);
   sq_for_every(&group->tg_members, curr)
     {
       tcb = container_of(curr, struct tcb_s, member);
 
       runtime += tcb->run_time;
     }
 
-  spin_unlock_irqrestore(NULL, flags);
+  spin_unlock_irqrestore(&group->lock, flags);
```

Comment:
```suggestion
  spin_unlock_irqrestore(&group->tg_members_lock, flags);
```

Reply:
This overhead is acceptable for performance improvement.

Reply:
are you really sure it improves performance? have you measured? otherwise, my gut feeling is that it's safer to avoid going too further into the "finer-grained lock is always better" assumption at this point.

Reply:
The code snippet you see is only protecting a specific resource, and fine-grained locks are worth it in this scenario because they avoid competing with other resources, and, spinlock does not take up too many memory. It is just a specific native prototype.

Reply:
well, it depends on many parameters. how frequently cpus actually compete on the specific lock(s) on a typical workload. what other data is on the cache line. how the specific hardware implements cache coherency and locks. etc, etc.
it's far from obvious to me.

don't get me wrong. i'm not saying it isn't worth. it might be.
i'm just saying it might not be a good idea to go too further w/o a benchmark.

Reply:
so let's do the finer tuning in the next after we get more performance data.